modules/chapter1/pages/section2.adoc

```diff
@@ -1,3 +1,355 @@
+The first mechanism is the Elyra Jupyter notebook addon which provides a visual editor for creating pipelines via orchestration of Jupyter notebooks. 
+
+The second mechanism and the one discussed here is using Data Science Pipelines (DSP) based on the upstream KubeflowPipelines V1. With DSP, pipelines are python scripts built using the KFPV1 python SDK. Once built the python pipeline is submitted to the DSP runtime to be scheduled for execution.
+
+[NOTE]
+====
+There is work in progress to migrate DSP to Kubeflow Pipeline V2. At the time of writing this is still a work in progress and we won't cover it here.
+====
+
+In OpenShift AI we use the *_Tekton_* runtime to execute pipelines. With this there is an additional step to be performed the python script needs to be compiled into a Tekton definition before being submitting to the runtime
+
+
+In OpenShift AI the Data Science Pipeline runtime consists of the following components
+
+* A Data Science Pipeline Server container a number of resources 
+* A MariaDB for storing pipeline definitions and results
+* A Pipeline scheduler for scheduling pipeline runs
+* A Persistent Agent to record the set of containers that executed as well as their inputs and outputs.
+
+As pipelines are executed multiple ephemeral pods will be created and terminated depending on the execution steps of the pipelines
+
+[NOTE]
+====
+DS Pipeline in OpenShift AI are managed by the 'data-science-pipelines-operator-controller-manager' operator in the redhat-ods-applications namespace. The CRD is an instance of _datasciencepipelinesapplications.datasciencepipelinesapplications.opendatahub.io_
+====
+
+.DataScience Pipeline pods
+****
+[source,cmd]
+----
+ ~/oc get pods
+NAME                                                              READY   STATUS    RESTARTS   AGE
+ds-pipeline-persistenceagent-pipelines-definition-6d6676d58vzss   1/1     Running   0          9m49s
+ds-pipeline-pipelines-definition-7c7dd56b4d-qfz8t                 2/2     Running   0          9m49s
+ds-pipeline-scheduledworkflow-pipelines-definition-7f5c645hcg4g   1/1     Running   0          9m48s
+mariadb-pipelines-definition-795c57795b-27ljh                     1/1     Running   0          9m49s
+----
+****
+
+== Creating a Data Science Pipeline with KFP
+
+=== Prerequisites 
+****
+* Python Setup
+[source,cmd]
+----
+pip install kfp==1.8
+pip install kfp-tekton==1.7.2
+----
+
+* S3 Storage
+
+Follow these instructions to setup a local Minio instance
+https://ai-on-openshift.io/tools-and-applications/minio/minio/
+****
+
+=== Creating a Pipeline Server
+
+To execute pipelines a _Pipeline server_ needs to be created but before that can happen an S3 Storage bucket needs to be created. This is used to store the run artifacts and outputs of any pipeline run on the associated server. 
+****
+image::creating-data-connection.png[]
+****
+Once the data connection is created then the pipeline server can be created
+****
+image::pre-creating-pipeline-server.png[]
```

Comment:
None of the images are rendering in the HTML. You have put the images along side the adoc. Please move all image files to a folder under chapter 1 called images, then all the images should render. Please see rhods-intro repo for example adoc files with correct references to images and videos.

Reply:
Also move all code that students need to clone into the rhod-qc-apps repo. Embedded code snippets can be put here and referenced in code call outs. Actual lab hands-on code should NOT live in this repo but moved to rhods-qc-apps. See rhods-intro repo for examples. You can embed markdown instructions and comments in the Jupyter notebooks themselves - again, see example notebook files in rhods-qc-apps repo under chapter 1 folder